### DISCUSSION TITLE
How to assign a unique graph id for each graph in the TUDataset?

### DISCUSSION THREAD

**Discussion by jiajunyu1999** [pyg-team/pytorch_geometric#7806] [Q&A] [answered]
I want to assign a unique graph id for each graph in the TUDataset as follows:
`
dataset = TUDataset(root = args.datapath, name = args.dataset)
dataset.data.graph_id = torch.arange(len(dataset))
dataset.graph_id = torch.arange(len(dataset))
`
However, I can add the graph id attribute for dataset.data, but cannot add it for dataset, the object of 'dataset' have only a fixed number of attributes? Because after feeding the object 'dataset' into 'dataloader', I want to make each batch has the graph id to identify the graph.
Thank you for your answer!

**Comment by akihironitta** [comment]
Hi @jiajunCAU, I'd pass `pre_transform` to the dataset constructor to add a new attribute to each data. I haven't tested this, but it'll look like this:

```python
class AddGraphIdTransform:
    def __init__(num_graphs):
        self.graph_id = 0
        self.num_graphs = num_graphs

    def __call__(self, data):
        data.graph_id = self.graph_id
        self.graph_id += 1
        return data

transform = AddGraphIdTransform(999)
dataset = TUDataset(..., pre_transform=transform)
```

**Reply by jiajunyu1999** [reply]
Thank you! But after testing, it seems don't work😭, the attributes in the dataset have not changed.

**Reply by EdisonLeeeee** [reply, marked as answer]
Need to delete your `processed/` dir and run it again.

**Reply by jiajunyu1999** [reply]
Yor're right! Thank you very much!!
